Update to syn-2

maurer opened the pull request:
Also moves a test out of doctest compilefail since syn-2 will build it correctly even without the "full" feature.

cuviper commented:
Do we use anything that would need "full" anymore? If not, I think we should just drop that feature. We can bump semver for the new MSRV requirement while we're at it, so a breaking feature change is ok.

cuviper commented on src/lib.rs:
```diff
@@ -170,16 +170,14 @@ impl NumTraits {
-                        panic!("#[num_traits] attribute value must be a str");
-                    }
+            if attr.path().is_ident("num_traits") {
+                if let syn::Expr::Lit(syn::ExprLit { lit: syn::Lit::Str(ref lit_str), .. }) = attr.meta.require_name_value().unwrap().value {
```
I'd rather this fell down to the panic message instead of `unwrap`.

maurer replied:
My bad, this one slipped through.

maurer commented:
I fixed up the `.unwrap()` call and added an additional commit to remove the `full-syntax` feature. There are still some things that are not present in `syn` without `full`, but none of them seem to correspond to complex discriminants.

glandium commented:
@cuviper is there anything holding this up?

cuviper commented:
bors r+

cuviper commented:
> is there anything holding this up?

Lack of time and attention... but it's published now!
